[v2.5.1] Application not able to access DB Service during startup

**Describe the bug**
We recently took the latest update of telepresence (v2.5.1) and tried it out with our cluster, but we are running into an issue when trying to launch our application locally. We have remote DBs running on Google Cloud SQL, and we have a kubernetes service that points to them. During our application startup, we are having environment properties that has the DB service endpoint, and we are making a DB connection using these properties.  We are able to run the intercept command without issues, and we are able to run `curl <db_service>.default` and get a response back from the server. However, when we launch our application locally with the environment properties pointing to the remote DB, we are making a connection pool, and the first 2 connections are working as expected, but we are getting errors when making the third consecutive connection.

We tried this process out with `2.4.11` and `2.5.1`, and we are only facing this issue with `2.5.1`.

I have attached the debug logs for 2.5.1 to this report:

[connector.log](https://github.com/telepresenceio/telepresence/files/8134292/connector.log)
[daemon.log](https://github.com/telepresenceio/telepresence/files/8134293/daemon.log)
[pod-1.namespace-1.log](https://github.com/telepresenceio/telepresence/files/8134294/pod-1.namespace-1.log)
[traffic-manager.namespace-2.log](https://github.com/telepresenceio/telepresence/files/8134295/traffic-manager.namespace-2.log)

**To Reproduce**
Steps to reproduce the behavior:
1. Create intercept by running the following command: `telepresence intercept component--port=10041 --env-file=/Users/suravikumar/rubikon/CLUSTER_NAME/component/component.env --preview-url=false --http-match=x-telepresence-intercept-id=suravikumar::.* --mount=/Users/suravikumar/rubikon/CLUSTER_NAME/component/mnt`
2. After intercept is successful, I am able to run `curl mysql.default` and receive an empty response (expected).
3. I then start the component locally with the DB details pointing to the Google Cloud SQL instance.
4. When component is starting up, it tries to make the DB connection, but we get the following exception:

```
2022-02-24 00:32:07.258 ERROR [component,,,] 4722 --- [main] o.a.t.j.p.ConnectionPool : [DBupgrade,,,DBupgrade,,,,,tenantName,,,] : Unable to create initial connections of pool.
com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.8.0_211]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:1.8.0_211]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_211]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[na:1.8.0_211]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:990) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:342) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2188) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2221) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2016) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:776) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.8.0_211]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:1.8.0_211]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_211]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[na:1.8.0_211]
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:386) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at org.apache.tomcat.jdbc.pool.PooledConnection.connectUsingDriver(PooledConnection.java:310) ~[tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.PooledConnection.connect(PooledConnection.java:203) ~[tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.ConnectionPool.createConnection(ConnectionPool.java:732) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.ConnectionPool.borrowConnection(ConnectionPool.java:664) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.ConnectionPool.init(ConnectionPool.java:479) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.ConnectionPool.<init>(ConnectionPool.java:154) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.pCreatePool(DataSourceProxy.java:118) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.createPool(DataSourceProxy.java:107) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.getConnection(DataSourceProxy.java:131) [tomcat-jdbc-8.5.11.jar!/:na]
	at org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource.getConnection(AbstractRoutingDataSource.java:164) [spring-jdbc-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at com.xxx.xxx.xxx.xxxxx.isTableExists(VersionHandler.java:531) [fw-dbupgrader-5.4.0-SNAPSHOT.jar!/:c066de4390b818f82bed303c8f67e60f17454c25+]
	at com.xxx.xxx.xxx.xxxxx.ifNoVersionTable(VersionHandler.java:525) [fw-dbupgrader-5.4.0-SNAPSHOT.jar!/:c066de4390b818f82bed303c8f67e60f17454c25+]
	at com.xxx.xxx.xxx.xxxxx.runDDLUpgrader(DBUpgrader.java:160) [fw-dbupgrader-5.4.0-SNAPSHOT.jar!/:c066de4390b818f82bed303c8f67e60f17454c25+]
	at com.xxx.xxx.xxx.xxxxx.postProcessAfterInitialization(DBUpgrader.java:127) [fw-dbupgrader-5.4.0-SNAPSHOT.jar!/:c066de4390b818f82bed303c8f67e60f17454c25+]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:423) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1633) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:585) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1264) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:372) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) [spring-beans-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.registerBeanPostProcessors(PostProcessorRegistrationDelegate.java:235) [spring-context-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.registerBeanPostProcessors(AbstractApplicationContext.java:702) [spring-context-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:527) [spring-context-4.3.7.RELEASE.jar!/:4.3.7.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:762) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:372) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:316) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1187) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1176) [spring-boot-1.4.5.RELEASE.jar!/:1.4.5.RELEASE]
	at com.xxx.xxx.xxx.xxxxx.main(ComponentServiceApplication.java:54) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_211]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_211]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [component-5.174.0-exec.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [component-5.174.0-exec.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [component-5.174.0-exec.jar:na]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) [component-5.174.0-exec.jar:na]
Caused by: java.net.ConnectException: Operation timed out (Connection timed out)
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[na:1.8.0_211]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[na:1.8.0_211]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[na:1.8.0_211]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[na:1.8.0_211]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.8.0_211]
	at java.net.Socket.connect(Socket.java:589) ~[na:1.8.0_211]
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:211) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:301) ~[mysql-connector-java-5.1.46.jar!/:5.1.46]
	... 74 common frames omitted
2022-02-24 00:32:07.278 WARN [component,,,] 4722 --- [main] o.s.b.c.e.AnnotationConfigEmbeddedWebApplicationContext : [DBupgrade,,,DBupgrade,,,,,tenantName] : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'archiveOrPurgeFulfillmentJobConfiguration': Unsatisfied dependency expressed through field 'dataSource'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSource': Initialization of bean failed; nested exception is com.xxx.xxx.xxx.xxxxx.RuntimeException: Error during DB Upgrade. Actual root cause: 
<OriginalException>
<FileName>PlainSocketImpl.java</FileName>
<LineNumber>-2</LineNumber>
<Message>java.net.ConnectException: Operation timed out (Connection timed out)</Message>
<DetailedMessage>Operation timed out (Connection timed out)</DetailedMessage>
<StackTrace>
java.net.ConnectException: Operation timed out (Connection timed out)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:211)
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:301)
	at com.mysql.jdbc.ConnectionImpl.coreConnect(ConnectionImpl.java:2188)
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2221)
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2016)
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:776)
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:386)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330)
	at org.apache.tomcat.jdbc.pool.PooledConnection.connectUsingDriver(PooledConnection.java:310)
	at org.apache.tomcat.jdbc.pool.PooledConnection.connect(PooledConnection.java:203)
	at org.apache.tomcat.jdbc.pool.ConnectionPool.createConnection(ConnectionPool.java:732)
	at org.apache.tomcat.jdbc.pool.ConnectionPool.borrowConnection(ConnectionPool.java:664)
	at org.apache.tomcat.jdbc.pool.ConnectionPool.init(ConnectionPool.java:479)
	at org.apache.tomcat.jdbc.pool.ConnectionPool.<init>(ConnectionPool.java:154)
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.pCreatePool(DataSourceProxy.java:118)
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.createPool(DataSourceProxy.java:107)
	at org.apache.tomcat.jdbc.pool.DataSourceProxy.getConnection(DataSourceProxy.java:131)
	at org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource.getConnection(AbstractRoutingDataSource.java:164)
	at com.xxx.xxx.xxx.xxxxx.isTableExists(VersionHandler.java:531)
	at com.xxx.xxx.xxx.xxxxx.ifNoVersionTable(VersionHandler.java:525)
	at com.xxx.xxx.xxx.xxxxx.runDDLUpgrader(DBUpgrader.java:160)
	at com.xxx.xxx.xxx.xxxxx.postProcessAfterInitialization(DBUpgrader.java:127)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:423)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1633)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:585)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1264)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:372)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.registerBeanPostProcessors(PostProcessorRegistrationDelegate.java:235)
	at org.springframework.context.support.AbstractApplicationContext.registerBeanPostProcessors(AbstractApplicationContext.java:702)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:527)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:762)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:372)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:316)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1187)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1176)
	at com.xxx.xxx.xxx.xxxxx.main(ComponentServiceApplication.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
</StackTrace>
</OriginalException>

```

**Expected behavior**
In version 2.4.11 of the application we are able to launch the application locally with a connection to our DB running in Google Cloud. We are expecting this to have the same behavior in later versions of telepresence

**Versions (please complete the following information):**
 - Telepresence Version: `v2.5.1 (api v3)`
 - Intel MacBook Pro 2018 - `Big Sur 11.6.4`, but I am also hearing from my team that this is happening for M1 Macs as well.
 - Google Kubernetes Engine -  `1.19.15-gke.1801`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 - Feature: Client-side binaries for the arm64 architecture are now available for linux
 
+- Bugfix: Fixed bug in the TCP stack causing timeouts after repeated connects to the same address
+
 ### 2.5.2 (February 23, 2022)
 
 - Bugfix: Fixed a bug where Telepresence would use the last server in resolv.conf

--- a/integration_test/multiple_services_test.go
+++ b/integration_test/multiple_services_test.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"math/rand"
 	"net/http"
 	"strings"
 	"sync"
@@ -100,6 +101,30 @@ func (s *multipleServicesSuite) Test_ListOnlyMapped() {
 
 	stdout = itest.TelepresenceOk(ctx, "list", "--namespace", s.AppNamespace())
 	require.NotContains(stdout, "No Workloads (Deployments, StatefulSets, or ReplicaSets)")
+}
+
+func (s *multipleServicesSuite) Test_RepeatedConnect() {
+	ctx := s.Context()
+	for i := 0; i < s.ServiceCount(); i++ {
+		url := fmt.Sprintf("http://%s-%d.%s", s.Name(), i, s.AppNamespace())
+		for v := 0; v < 30; v++ {
+			s.Run(fmt.Sprintf("test-%d", i*30+v), func() {
+				s.T().Parallel()
+				time.Sleep(time.Duration(rand.Intn(100)) * time.Millisecond)
+				assert := s.Assert()
+				cl := http.Client{}
+				req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+				req.Close = true
+				assert.NoError(err)
+				res, err := cl.Do(req)
+				assert.NoError(err)
+				assert.Equal(res.StatusCode, http.StatusOK)
+				_, err = io.Copy(io.Discard, res.Body)
+				assert.NoError(err)
+				assert.NoError(res.Body.Close())
+			})
+		}
+	}
 }
 
 func (s *multipleServicesSuite) Test_ProxiesOutboundTraffic() {

--- a/pkg/client/rootd/router.go
+++ b/pkg/client/rootd/router.go
@@ -183,9 +183,9 @@ func (s *session) tcp(c context.Context, pkt tcp.Packet) {
 		return
 	}
 
-	wf, _, err := s.handlers.GetOrCreateTCP(c, connID, func(c context.Context, remove func()) (tunnel.Handler, error) {
+	wf, _, err := s.handlers.GetOrCreate(c, connID, func(c context.Context, remove func()) (tunnel.Handler, error) {
 		return tcp.NewHandler(s.streamCreator(connID), &s.closing, vifWriter{s.dev}, connID, remove, s.rndSource), nil
-	}, pkt)
+	})
 	if err != nil {
 		dlog.Error(c, err)
 		pkt.Release()

--- a/pkg/tunnel/pool.go
+++ b/pkg/tunnel/pool.go
@@ -4,32 +4,13 @@ import (
 	"context"
 	"errors"
 	"sync"
-	"time"
 
 	"github.com/datawire/dlib/dlog"
-	"github.com/telepresenceio/telepresence/v2/pkg/vif/ip"
 )
-
-// RecursionBlocker is implemented by handlers that may experience recursive calls
-// back into the TUN device for IP addresses that have not been forwarded by the cluster.
-// This typically happens when running a cluster in a docker container on the local host
-// and making attempts to connect to an existing IP on a port that no service is
-// listening to.
-type RecursionBlocker interface {
-	InitDone() <-chan struct{}
-	Proceed() bool
-	Reset(context.Context, ip.Packet) error
-	Discard(ip.Packet) bool
-}
-
-// The error returned when recursion is encountered
-var errRecursion = errors.New("connection recursion")
 
 type Pool struct {
 	handlers map[ConnID]Handler
-	blockers map[ip.AddrKey]RecursionBlocker
-
-	lock sync.RWMutex
+	lock     sync.RWMutex
 }
 
 type Handler interface {
@@ -40,31 +21,12 @@ type Handler interface {
 }
 
 func NewPool() *Pool {
-	return &Pool{handlers: make(map[ConnID]Handler), blockers: make(map[ip.AddrKey]RecursionBlocker)}
+	return &Pool{handlers: make(map[ConnID]Handler)}
 }
 
 func (p *Pool) release(ctx context.Context, id ConnID) {
 	p.lock.Lock()
-	if h, ok := p.handlers[id]; ok {
-		delete(p.handlers, id)
-		if b, ok := h.(RecursionBlocker); ok {
-			destKey := ip.MakeAddrKey(id.Destination(), id.DestinationPort())
-			if p.blockers[destKey] == b {
-				if !b.Proceed() {
-					// Delete after a delay to ensure that all recursive attempts to
-					// establish have ceased.
-					time.AfterFunc(20*time.Second, func() {
-						p.lock.Lock()
-						delete(p.blockers, destKey)
-						p.lock.Unlock()
-					})
-				} else {
-					// Delete now (using the current lock)
-					delete(p.blockers, destKey)
-				}
-			}
-		}
-	}
+	delete(p.handlers, id)
 	count := len(p.handlers)
 	p.lock.Unlock()
 	dlog.Debugf(ctx, "-- POOL %s, count now is %d", id, count)
@@ -112,71 +74,6 @@ func (p *Pool) GetOrCreate(ctx context.Context, id ConnID, createHandler Handler
 	var old Handler
 	if old, ok = p.handlers[id]; !ok {
 		p.handlers[id] = handler
-	}
-	count := len(p.handlers)
-	p.lock.Unlock()
-	if ok {
-		// Toss newly created handler. It's not started anyway.
-		return old, true, nil
-	}
-	handler.Start(handlerCtx)
-	dlog.Debugf(ctx, "++ POOL %s, count now is %d", id, count)
-	return handler, false, nil
-}
-
-// GetOrCreateTCP is like GetOrCreate but with the addition that it detects and delays attempts to
-// create handlers for the same destination IP and port until the first attempt has either succeeded
-// or failed. If it fails, then attempts made during between the start and end of that attempt will
-// fail too.
-//
-func (p *Pool) GetOrCreateTCP(ctx context.Context, id ConnID, createHandler HandlerCreator, initialPacket ip.Packet) (Handler, bool, error) {
-	var blocker RecursionBlocker
-	p.lock.RLock()
-	if handler, ok := p.handlers[id]; ok {
-		p.lock.RUnlock()
-		return handler, true, nil
-	}
-
-	blocker = p.blockers[ip.MakeAddrKey(id.Destination(), id.DestinationPort())]
-	if blocker != nil {
-		p.lock.RUnlock()
-		if blocker.Discard(initialPacket) {
-			return nil, false, nil
-		}
-		<-blocker.InitDone()
-		if blocker.Proceed() {
-			return p.GetOrCreate(ctx, id, createHandler)
-		}
-		if err := blocker.Reset(ctx, initialPacket); err != nil {
-			dlog.Errorf(ctx, "failed to send RST after recursion block: %v", err)
-		}
-		return nil, false, errRecursion
-	}
-
-	handlerCtx, cancel := context.WithCancel(ctx)
-	release := func() {
-		p.release(ctx, id)
-		cancel()
-	}
-
-	handler, err := createHandler(handlerCtx, release)
-	p.lock.RUnlock()
-	if err != nil {
-		return nil, false, err
-	}
-	if handler == nil {
-		return nil, false, errors.New("createHandler function did not produce a handler")
-	}
-
-	p.lock.Lock()
-	var old Handler
-	var ok bool
-	if old, ok = p.handlers[id]; !ok {
-		p.handlers[id] = handler
-		var isBlocker bool
-		if blocker, isBlocker = handler.(RecursionBlocker); isBlocker {
-			p.blockers[ip.MakeAddrKey(id.Destination(), id.DestinationPort())] = blocker
-		}
 	}
 	count := len(p.handlers)
 	p.lock.Unlock()

--- a/pkg/vif/tcp/handler.go
+++ b/pkg/vif/tcp/handler.go
@@ -101,12 +101,6 @@ type handler struct {
 	toTun   ip.Writer
 	fromTun chan Packet
 
-	// recursionWaiter blocks connects to the same destination during connect to prevent endless
-	// recursions when the destination runs on the local host (in a Docker container) and forwards
-	// connects that aren't routed to Telepresence TUN-device
-	establishedCh chan struct{}
-	establishedOk bool
-
 	// the dispatcher signals its intent to close in dispatcherClosing. 0 == running, 1 == closing, 2 == closed
 	dispatcherClosing *int32
 
@@ -180,8 +174,6 @@ type handler struct {
 
 	// random generator for initial sequence number
 	rnd *rand.Rand
-
-	synDiscardCount int
 }
 
 func NewHandler(
@@ -201,7 +193,6 @@ func NewHandler(
 		fromTun:           make(chan Packet, ioChannelSize),
 		toMgrCh:           make(chan Packet, ioChannelSize),
 		toMgrMsgCh:        make(chan tunnel.Message),
-		establishedCh:     make(chan struct{}),
 		myWindow:          maxReceiveWindow,
 		wfState:           stateIdle,
 		rnd:               rand.New(rndSource),
@@ -232,34 +223,11 @@ func (h *handler) Close(ctx context.Context) {
 	}
 	// Wake up if waiting for larger window size (ends processPayload)
 	h.sendCondition.Broadcast()
-	select {
-	case <-h.establishedCh:
-		// already closed
-	default:
-		close(h.establishedCh)
-	}
-}
-
-func (h *handler) InitDone() <-chan struct{} {
-	return h.establishedCh
-}
-
-func (h *handler) Proceed() bool {
-	return h.establishedOk // This isn't a state. It's a permanent indication that the connection attempt succeeded.
 }
 
 // Reset replies to the sender of the initialPacket with a RST packet.
 func (h *handler) Reset(ctx context.Context, initialPacket ip.Packet) error {
 	return h.toTun.Write(ctx, initialPacket.(Packet).Reset())
-}
-
-// Discard returns true if the package should be discarded
-func (h *handler) Discard(initialPacket ip.Packet) bool {
-	if initialPacket.(Packet).Header().SYN() {
-		h.synDiscardCount++
-		return h.synDiscardCount > 1 // Keep initial SYN packet
-	}
-	return false
 }
 
 func (h *handler) Start(ctx context.Context) {

--- a/pkg/vif/tcp/tmgr.go
+++ b/pkg/vif/tcp/tmgr.go
@@ -12,8 +12,6 @@ import (
 func (h *handler) handleStreamControl(ctx context.Context, ctrl tunnel.Message) {
 	switch ctrl.Code() {
 	case tunnel.DialOK:
-		h.establishedOk = true
-		close(h.establishedCh)
 	case tunnel.DialReject, tunnel.Disconnect:
 		h.Close(ctx)
 	case tunnel.KeepAlive:

